cli/main.rs

```diff
@@ -19,24 +20,42 @@ enum Mode {
         /// The base to use for encoding.
         #[structopt(short = "b", long = "base", default_value = "base58btc")]
         base: StrBase,
-        /// The data need to be encoded.
+        /// The data to encode. Reads from stdin if not provided.
         #[structopt(short = "i", long = "input")]
-        input: String,
+        input: Option<String>,
     },
     #[structopt(name = "decode")]
     Decode {
-        /// The data need to be decoded.
+        /// The data to decode. Reads from stdin if not provided.
         #[structopt(short = "i", long = "input")]
-        input: String,
+        input: Option<String>,
     },
 }
 
 fn main() -> Result<()> {
     env_logger::init();
     let opts = Opts::from_args();
     match opts.mode {
-        Mode::Encode { base, input } => encode(base, input.as_bytes()),
-        Mode::Decode { input } => decode(&input),
+        Mode::Encode { base, input } => {
+            let input_bytes = if let Some(s) = input {
+                s.into_bytes()
+            } else {
+                let mut buf = Vec::new();
+                io::stdin().read_to_end(&mut buf)?;
+                buf
+            };
+            encode(base, &input_bytes)
+        }
+        Mode::Decode { input } => {
+            let input_str = if let Some(s) = input {
+                s
+            } else {
+                let mut buf = String::new();
+                io::stdin().read_to_string(&mut buf)?;
+                buf
+            };
```

Comment:
Same reasons as for the comment above:

```suggestion
            let input_str = match input {
                Some(s) => s,
                None => {
                    let mut buf = String::new();
                    io::stdin().read_to_string(&mut buf)?;
                    buf
                }
            };
```